sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-strawman-october-2019.md

```diff
@@ -0,0 +1,536 @@
+
+- Exposes the `PartitionContext` that identifies the partition from which the `EventData` was received.
+
+- Offers a method to create/update a checkpoint based on the event for the associated partition.
+
+### Event Hub Connection
+
+- Represents a single connection to the Event Hubs service for a specific Event Hub.
+
+- May be used to create a producer or a consumer, ensuring that developers have transparency into the number of connections used and have control over their re-use.
+
+- Created implicitly by a producer or consumer when not provided at construction.
+
+- May be closed/disposed, which closes all types sharing the connection.
+
+### Event Processor Client Base
+
+- Represents a base-level type for cooperative event processing, providing functionality for load balancing and dispatching events for processing. 
+
+- Does not offer the higher-level abstractions of the opinionated client; operations are performed by overriding/implementing base class methods rather than using an event-based model.  The arguments for these methods are built around the set of lower-level types and do not make use of the `EventArg` abstractions.
+
+- Exposes events in batches, rather than single events; the batch size and prefetch count support configuration in order to allow for optimal tuning.
+
+- Guarantees that only one event-batch-per-partition is dispatched for processing at one time and manages concurrency to allow multiple partitions to be processing at once.
+
+- Abstract; meant to be extended by developers with advanced customization needs.
+
+## Publishing scenarios 
+
+### Create a Producer Client
+
+```csharp
+// Event Hub connection string
+var connectionString = "<< EVENT HUB CONNECTION STRING FROM PORTAL >>";
+var producer = new EventHubProducerClient(connectionString);
+
+// Namespace connection string and Event Hub name
+var connectionString = "<< NAMESPACE CONNECTION STRING FROM PORTAL >>";
+var producer = new EventHubProducerClient(connectionString, "Event Hub Name");
+
+// Expanded form
+var fqNamespace = "Fully Qualified Event Hubs Namespace";
+var hub = "Event Hub Name";
+var tokenCredential = GetCredentialFromIdentityClient(...);
+var producer = new EventHubProducerClient(fqNamespace, hub, tokenCredential);
+
+// Existing connection
+var connection = new EventHubsConnection(...);
+var producer = new EventHubProducerClient(connection);
+```
+
+### Send with Automatic partition routing
+
+```csharp
+var producer = BuildProducerClient(...);
+
+var batch = await producer.CreateBatchAsync();
+AddEventsToBatch(batch);
+
+await producer.SendAsync(batch);
+```
+
+### Send using a partition key
+
+```csharp
+var producer = BuildProducerClient(...);
+
+var batchOptions = new BatchOptions { PartitionKey = "myKey" };
+var batch = await producer.CreateBatchAsync(batchOptions);
+AddEventsToBatch(batch);
+
+await producer.SendAsync(batch);
+```
+
+### Send to a specific partition
```

Comment:
Is this intended to be distinct from the previous example, or is it duplicate?

Reply:
They're different, albeit very subtly.   The options specify a `PartitionKey` in one and a `PartitionId` in the other.